### PR TITLE
[openssl] add feature to enable ssl3/weak-ssl-ciphers

### PR DESCRIPTION
**Pull request by BamButz**
I removed the changes made in #28690 because they are the default values (see here https://github.com/openssl/openssl/blob/master/Configure#L557).

In addition, I was in need to enable these functionalities, so I added features to enable them when needed.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/openssl/portfile.cmake [hunk 1]
```diff
@@ -56,6 +54,15 @@ if(NOT "tools" IN_LIST FEATURES)
     vcpkg_list(APPEND CONFIGURE_OPTIONS no-apps)
 endif()
 
+if("weak-ssl-ciphers" IN_LIST FEATURES)
+	vcpkg_list(APPEND CONFIGURE_OPTIONS enable-weak-ssl-ciphers)
```
Comment (dg0yt) [inline ports/openssl/portfile.cmake:58]:
```suggestion
    vcpkg_list(APPEND CONFIGURE_OPTIONS enable-weak-ssl-ciphers)
```

File: ports/openssl/portfile.cmake [hunk 2]
```diff
@@ -56,6 +54,15 @@ if(NOT "tools" IN_LIST FEATURES)
...
+endif()
+
+if("ssl3" IN_LIST FEATURES)
+	vcpkg_list(APPEND CONFIGURE_OPTIONS enable-ssl3)
+	vcpkg_list(APPEND CONFIGURE_OPTIONS enable-ssl3-method)
```
Comment (dg0yt) [inline ports/openssl/portfile.cmake:63]:
```suggestion
    vcpkg_list(APPEND CONFIGURE_OPTIONS enable-ssl3)
    vcpkg_list(APPEND CONFIGURE_OPTIONS enable-ssl3-method)
```

**Review by dg0yt — CHANGES_REQUESTED** [review]
Just for consistent formatting: no tabs.

**Comment by BamButz** [comment]
Thanks, I've updated my code formatting settings.

**Comment by BamButz** [comment]
Is there something left to do for me? I would really appreciate this to be merged.

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
All features are tested successfully in the following triplet:

1. x86-windows
2. x64-windows
3. x64-windows-stataic

File: ports/openssl/portfile.cmake [hunk 3]
```diff
@@ -58,6 +56,15 @@ if(NOT "tools" IN_LIST FEATURES)
     vcpkg_list(APPEND CONFIGURE_OPTIONS no-apps)
 endif()
 
+if("weak-ssl-ciphers" IN_LIST FEATURES)
+    vcpkg_list(APPEND CONFIGURE_OPTIONS enable-weak-ssl-ciphers)
```
Comment (BillyONeal) [inline ports/openssl/portfile.cmake:60]: For consistency shouldn't these also set `no-weak-ssl-ciphers` and/or `no-ssl3`?

**Review by BillyONeal — COMMENTED** [review]
Is it a good idea to add a feature that presumably should never be used any longer?

(I'm not necessarily opposed to adding it, but I think we at least need on record a reason why we were in a secure condition and decided to allow an insecure condition)

**Comment by BillyONeal** [comment]
Put another way: It seems likely to me that this feature was intentionally not added before, so I think we need a reason on record as to why we changed our minds.

File: ports/openssl/portfile.cmake (hunk 3 above)
Comment (BamButz) [reply ports/openssl/portfile.cmake:60]: no-weak-ssl-ciphers and no-ssl3 are default values, they do not need to be set.

**Comment by BamButz** [comment]
The reason I need this is that I'm working on internal legacy software, which is not supposed to be changed to another cipher. I understand your concerns and that's why I implemented this change as a feature so you have to opt in to enable the insecure ciphers.

**Comment by BillyONeal** [comment]
OK, just be aware that actively exploited attacks like CRIME and POODLE are out there, and that SSL 3 has been obsolete for 24 years, and broken for 9 years. There is pretty much no use case that would require encryption that are OK with SSL 3 anymore.

Nonetheless, it is an option exposed by upstream, so I'll merge it.

**Pull request merged**